src/modules/phishing-defence/phishing-defence-service.ts

```diff
@@ -97,6 +97,9 @@ class PhishingDefence {
       };
     }
     const origin = url ? this.getSafeOrigin(url) : null;
+    if (origin && this.websiteStatus[origin] === 'error') {
+      await this.checkDapp(url);
+    }
```

Comment:
Can you please explain these lines? I don't understand at all what this does and why the condition is like this

Reply:
Sure, let me update the pr and provide all context

Reply:
Thanks!

> // to provide the correct data after passing the captcha we need to check the status again

what is "correct data" here?

Reply:
Hm. The actual data we have about the dapp, not error because of cloudflare block.

Reply:
I finally figured out what this does.

So after completing the captcha we call "refetchQueries", and in case of `<PhishingDefenceStatus />` being open, the refetching will get old data because the result is outdated in `phishingDefenceService`.

**So this code serves as a "retry" in case of an error result**

 I guess there are other ways to achieve this but perhaps it's okay to leave it as is for now
 
 Ideally though request statuses are better organized by some caching lib